src/main/java/me/arminb/sara/controller/QuestionController.java

```diff
@@ -0,0 +1,107 @@
+            } else {
+                return new ResponseEntity<Question>(HttpStatus.NOT_FOUND);
+            }
+        }
+        catch (DataAccessException e){
+            return new ResponseEntity<Question>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @RequestMapping(value="/questions/search",method = RequestMethod.GET)
+    public  ResponseEntity<List<Question>> find( @RequestParam(value="title", required=false) String title,
+                                             @RequestParam(value="tag", required = false) String tag,
+                                             @RequestParam(value="page", required = false) Integer pageNumber,
+                                             @RequestParam(value="pageCount", required = false) Integer pageCount) {
+        try {
+            List<Question> questions = questionService.find(title, tag, pageNumber, pageCount);
+            if(questions != null) {
+                return new ResponseEntity<List<Question>>(questions, HttpStatus.OK);
+            } else {
+                return new ResponseEntity<List<Question>>(HttpStatus.NOT_FOUND);
+
+            }
+        }
+        catch (DataAccessException e){
+            return new ResponseEntity<List<Question>>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @RequestMapping(value="/questions",method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public ResponseEntity<Question> addQuestion(@RequestBody QuestionRequest questionReq) {
+        try {
+            Question question = questionReq.toQuestion();
+            return new ResponseEntity<Question>(questionService.saveQuestion(question), HttpStatus.OK);
+        }
+        catch (DataAccessException e){
+            return new ResponseEntity<Question>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @RequestMapping(value= "questions/{id}", method = RequestMethod.PUT, consumes =  MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public ResponseEntity<Question> updateQuestion(@PathVariable("id") String id, @RequestBody QuestionRequest questionReq) {
+        try {
+            Question question = questionReq.toQuestion();
+            question.setId(id);
+            Question question_obj = questionService.saveQuestion(question);
+            if(question_obj != null) {
+                return new ResponseEntity<Question>(question_obj, HttpStatus.OK);
+            } else {
+                return new ResponseEntity<Question>(HttpStatus.NOT_FOUND);
+            }
+        }
+        catch (DataAccessException e){
+            return new ResponseEntity<Question>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @RequestMapping(value="questions/{id}", method = RequestMethod.DELETE)
+    public ResponseEntity<Boolean> deleteQuestion(@PathVariable("id") String id) {
+        try {
+            return new ResponseEntity<Boolean>(questionService.deleteQuestion(id), HttpStatus.OK);
```

Comment:
what if id is not found ? You should return 404 in that case. I would do the same thing for user deletion in UserController